unittests/test_cli.py

```diff
@@ -281,6 +281,51 @@ def test_checkpath_colon_separated(self):
             r'Found (\d+) check', stdout, re.MULTILINE).group(1)
         self.assertEqual(num_checks, '2')
 
+    def test_checkpath_duplicate(self):
```

Comment:
The whole functionality about removing duplicates should be put in a separate `os_ext` utility function and tested separately. Invoking ReFrame for testing such a stand-alone feature is overkill. Also in terms of test code + execution time.